plugins/modules/swim_workflow_manager.py

```diff
@@ -1096,24 +1303,35 @@ def get_diff_import(self):
                 return self
 
             if self.want.get("import_type") == "remote":
-                import_payload_dict = {}
-                temp_payload = self.want.get("url_import_details").get("payload")[0]
-                keys_to_change = list(import_key_mapping.keys())
+                import_image_payload = []
+                temp_payloads = self.want.get("url_import_details").get("payload")
 
-                for key, val in temp_payload.items():
-                    if key in keys_to_change:
-                        api_key_name = import_key_mapping[key]
-                        import_payload_dict[api_key_name] = val
+                for temp_payload in temp_payloads:
+                    source_urls = temp_payload.get('source_url', [])
+
+                    for url in source_urls:
+                        import_payload_dict = {}
+
+                        if 'source_url' in import_key_mapping:
+                            import_payload_dict['sourceURL'] = url
+
+                        if 'image_family' in import_key_mapping:
+                            import_payload_dict['imageFamily'] = temp_payload.get('image_family')
+
+                        if 'application_type' in import_key_mapping:
+                            import_payload_dict['applicationType'] = temp_payload.get('application_type')
```

Comment:
These 'image_family' and 'application_type' will be same for all devices so better to keep it outside the loop.

Reply:
we have to pass separately as yesterday discussed on the call